app/src/lib/stores/git-store.ts

```diff
@@ -372,6 +385,37 @@ export class GitStore extends BaseStore {
         .shift() || null
   }
 
+  public async fetchTagsToPush(account: IGitAccount | null) {
+    const currentRemote = this._currentRemote
+
+    if (currentRemote === null) {
+      this._tagsToPush = null
+      return
+    }
+
+    if (this.tip.kind !== TipState.Valid) {
+      this._tagsToPush = null
+      return
+    }
+    const currentBranch = this.tip.branch
+
+    const tagsToPush = await this.performFailableOperation(async () => {
+      const localTags = await this.getAllTags()
+
+      return fetchTagsToPushMemoized(
+        this.repository,
+        account,
+        currentRemote,
+        currentBranch.name,
+        localTags,
+        currentBranch.tip.sha
+      )
+    })
+    this._tagsToPush = tagsToPush !== undefined ? tagsToPush : null
+
+    this.emitUpdate()
```

Comment:
do we need to emit an update if the list of tags to push doesn't change?

Reply:
No we don't 😄 I'll change that 🤡